src/node_constants.cc

```diff
@@ -1288,41 +1289,24 @@ void CreatePerContextProperties(Local<Object> target,
   CHECK(
       target->SetPrototypeV2(env->context(), Null(env->isolate())).FromJust());
 
-  Local<Object> os_constants = Object::New(isolate);
-  CHECK(os_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> err_constants = Object::New(isolate);
-  CHECK(err_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> sig_constants = Object::New(isolate);
-  CHECK(sig_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> priority_constants = Object::New(isolate);
-  CHECK(priority_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> fs_constants = Object::New(isolate);
-  CHECK(fs_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> crypto_constants = Object::New(isolate);
-  CHECK(crypto_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> zlib_constants = Object::New(isolate);
-  CHECK(zlib_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> dlopen_constants = Object::New(isolate);
-  CHECK(dlopen_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
-
-  Local<Object> trace_constants = Object::New(isolate);
-  CHECK(trace_constants->SetPrototypeV2(env->context(), Null(env->isolate()))
-            .FromJust());
+  Local<Object> os_constants =
+      Object::New(isolate, Null(isolate), nullptr, nullptr, 0);
```

Comment:
You could probably defer the creation of `os_constants` and use a `v8::LocalVector` to collect the individual child objects here and create the `os_constants` after passing in those values instead of the `nullptr`s here.

Reply:
I'm planning on doing a follow up for all those cases. I wanted to optimize the unnecessary SetPrototypeV2 calls first